src/windows_native/descriptor/typedefs.rs

```diff
@@ -0,0 +1,197 @@
+use std::mem::size_of;
+use crate::windows_native::descriptor::types::BitRange;
+
+#[macro_export]
+macro_rules! const_assert {
+    ($x:expr $(,)?) => {
+        #[allow(unknown_lints)]
+        const _: [(); 0 - !{ const ASSERT: bool = $x; ASSERT } as usize] = [];
+    };
+}
+
+pub type Usage = u16;
+
+const_assert!(size_of::<LinkCollectionNode>() == 16);
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct LinkCollectionNode {
+    pub link_usage: Usage,
+    pub link_usage_page: Usage,
+    pub parent: u16,
+    pub number_of_children: u16,
+    pub next_sibling: u16,
+    pub first_child: u16,
+    pub bits: u32
+}
+
+impl LinkCollectionNode {
+    pub fn is_alias(&self) -> bool {
+        self.bits & 1u32 << 8 != 0
+    }
+    pub fn collection_type(&self) -> u8 {
+        (self.bits & 0xFFu32) as u8
+    }
+}
+
+const_assert!(size_of::<CapsInfo>() == 8);
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct CapsInfo {
+    pub first_cap: u16,
+    pub number_of_caps: u16,
+    pub last_cap: u16,
+    pub report_byte_length: u16
+}
+
+const_assert!(size_of::<UnknownToken>() == 8);
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct UnknownToken {
+    pub token: u8,
+    _reserved: [u8; 3],
+    pub bit_field: u32
+}
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct Button {
+    pub logical_min: i32,
+    pub logical_max: i32
+}
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct NotButton {
+    pub has_nul: u8,
+    _reserved: [u8; 3],
+    pub logical_min: i32,
+    pub logical_max: i32,
+    pub physical_min: i32,
+    pub physical_max: i32
+}
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+union MaybeButton {
+    button: Button,
+    not_button: NotButton
+}
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct Range {
+    pub usage_min: Usage,
+    pub usage_max: Usage,
+    pub string_min: u16,
+    pub string_max: u16,
+    pub designator_min: u16,
+    pub designator_max: u16,
+    pub data_index_min: u16,
+    pub data_index_max: u16
+}
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct NotRange {
+    pub usage: Usage,
+    _reserved1: Usage,
+    pub string_index: u16,
+    _reserved2: u16,
+    pub designator_index: u16,
+    _reserved3: u16,
+    pub data_index: u16,
+    _reserved4: u16,
+}
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+union MaybeRange {
+    range: Range,
+    not_range: NotRange
+}
+
+
+const_assert!(size_of::<Caps>() == 104);
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct Caps {
```

Comment:
Would be great to add some comments from where exactly these definitions are taken (e.g. msdn page, etc...).